### DISCUSSION TITLE
Support for JSON based experiment schema?

### DISCUSSION THREAD

**Discussion by veritymedia** [jspsych/jsPsych#2514] [Q&A]
First, I want to say thanks for creating this library, very awesome!

I must admit that I am still quite new to this library, so excuse my potential ignorance. I have read a large part of the docs, but I cant find any info on loading (& running) a study from JSON . Many of the features seem amenable to being stored in such a format (eg. timelines, timeline variables, etc), but only in their simpler forms. More interesting things like dynamic parameters would be limited (am i wrong?).

**Comment by jodeleeuw** [comment]
Hi @veritymedia,

There's no direct support for loading a JSON file. You could write a timeline in JSON and then load it into a JS variable with `JSON.parse()`, and then run the timeline.

You're right that some features would be limited, like dynamic parameters.
